saleor/checkout/models.py

```diff
@@ -186,6 +239,100 @@ class CheckoutLine(models.Model):
     )
     quantity = models.PositiveIntegerField(validators=[MinValueValidator(1)])
 
+    currency = models.CharField(
+        max_length=settings.DEFAULT_CURRENCY_CODE_LENGTH,
+    )
+
+    undiscounted_unit_price_net_amount = models.DecimalField(
+        max_digits=settings.DEFAULT_MAX_DIGITS,
+        decimal_places=settings.DEFAULT_DECIMAL_PLACES,
+        default=0,
+    )
+    undiscounted_unit_price_gross_amount = models.DecimalField(
+        max_digits=settings.DEFAULT_MAX_DIGITS,
+        decimal_places=settings.DEFAULT_DECIMAL_PLACES,
+        default=0,
+    )
+    undiscounted_unit_price = TaxedMoneyField(
+        net_amount_field="undiscounted_unit_price_net_amount",
+        gross_amount_field="undiscounted_unit_price_gross_amount",
+    )
+    unit_price_net_amount = models.DecimalField(
+        max_digits=settings.DEFAULT_MAX_DIGITS,
+        decimal_places=settings.DEFAULT_DECIMAL_PLACES,
+        default=0,
+    )
+    unit_price_gross_amount = models.DecimalField(
```

Comment:
Would be better to have default=Decimal(0) here. Function like quantitze_price could fail as they will get int as an input. 
Also, question. Are you sure that we should set default values for these fields? I am not sure of this.

Reply:
Why not? All checkout prices will be calculated by manager methods, I don't think we should duplicate those calculations when eg. creating a new line.